### DISCUSSION TITLE
Digital Product Support - Pixel, GA4 Support - Comprehensive Description - Customer Review

### DISCUSSION THREAD

**Discussion by hfzi** [medusajs/medusa#4765] [Feature Requests]
Hello, I have 3 questions;
1. Will google analytics for ecommerce and facebook pixel support be added for the future? These tools are needed to follow sales and campaigns. There is no need to pay segment for this. 
2. Will there be digital product sales support? It is enough to share google cloud or aws s3 download link for users who purchase.
3. Digital products have descriptions, images, videos etc.; how can I add them to the description section like in gumroad, wordpress
4. How can I get a customer to comment and rate a customer?

Thanks, Regards.

**Comment by srindom** [comment]
1. You can add GA and FB pixel in a few ways: a. if you are doing client-side tracking simply add it to your storefront code, b. if you want to do server-side tracking you can add subscribers to the events you want to send along. You can see the Segment plugin for inspiration - the necessary change would be to send the event to the GA/FB APIs instead of Segment.
2. There are no plans to add digital products in Medusa's core. It is, however, fairly easy to implement the functionality through a plugin. A few weeks back I played around with this during a weekend. You can read more here: https://twitter.com/sebrindom/status/1673975677131935745?s=20
3. There is no straight-forward path for this at this point; one option is to treat the description field as markdown text and then format when you need to render.
4. Customer Reviews is something you can support through plugins. @pevey has built a cool plugin for this: https://github.com/pevey/medusa-plugin-ratings

**Comment by hfzi** [comment]
thank you very much. it is necessary to integrate event codes for google analytics 4 and facebook pixel e-commerce tracking. so that the target audience performance can be tracked from the dashboard of the platforms, which products are sold and added to the cart. the project is great, I will try it as soon as possible. the advanced description section can be nice, so that more effective and long descriptions can be written, and video / gif / link can be added with html support. i will try the plugin, good luck
